tutorials/build_system/gradle_guide.md

```diff
@@ -0,0 +1,225 @@
+
+The storage location of downloaded IDE versions and components defaults to the Gradle cache directory.
+However, it can be controlled by setting the `intellij.ideaDependencyCachePath` attribute.
+
+### Controlling Downloads by the Gradle Plugin
+As mentioned in the section about [configuring the intellij platform](#configuring-the-intellij-platform-used-for-building-plugin-projects) used for building plugin projects, the Gradle plugin will fetch the version of the IntelliJ Platform specified by the default or by the `intellij` attributes.
+Whether to download source code is controlled by the attribute `intellij.downloadSources`. If source code isn't needed, then this is one download (and indexing) step that can be saved.
+
+Standardizing the versions of the Gradle plugin and Gradle system across projects will minimize the time spent downloading versions.
+There are controls for managing the IntelliJ IDEA Gradle plugin version, and the version of Gradle itself.
+The Gradle plugin version is defined in the `plugins {}` section of a project's `build.gradle` file.
+The Gradle version is in defined in a project's `gradle-wrapper.properties`.
+
+### Patching the Plugin.XML File
+With the exception of the `since-build` and `until-build` attributes of the `<idea-version>` element, all of the substitution attributes in the Patching DSL will be patched into the corresponding element values in a plugin project's `plugin.xml` file.
+The `since-build` and `until-build` substitution can be inhibited by setting `intellij.updateSinceUntilBuild` to `false`.
+Substitution of at least the default value for the other attributes cannot be inhibited.
+
+A best practice to avoid confusion is to replace the elements in `plugin.xml` that will be patched by the Gradle plugin with a comment.
+That way the values for these parameters do not appear two places in the source code.
+The Gradle plugin will add the necessary elements as part of the patching process.
+For those attributes that contain descriptions such as `changeNotes` and `pluginDescription`, a `CDATA` block is not necessary when using HTML elements.
+
+As discussed in [Components of a Wizard-Generated Gradle IntelliJ Platform Plugin](prerequisites.html#components-of-a-wizard-generated-gradle-intellij-platform-plugin), `project.version`, `project.group`, and `rootProject.name` are all generated based on the input to the Wizard.
+However, the IntelliJ IDEA Gradle plugin does not combine and substitute those values for the default `<id>` and `<name>` elements in the `plugin.xml` file.
+There is no IntelliJ Platform-related reason the value of those XML elements must match the Gradle plugin values, but it is considered a best practice to avoid confusion.
+
+Smaller projects with only one `plugin.xml` file will likely use the defaults for which file to patch and where to store the patched file.
```

Comment:
I'd avoid the whole paragraph about having multiple `plugin.xml`, as it really is a _very_ rare corner case

Reply:
Gone!